frontend/components/forms/admin/AppConfigForm/AppConfigFormFunctional.tsx

```diff
@@ -0,0 +1,1121 @@
+    // SAML single sign on options
+    enableSSO: formData.sso_settings.enable_sso || false,
+    idpName: formData.sso_settings.idp_name || "",
+    entityID: formData.sso_settings.entity_id || "",
+    issuerURI: formData.sso_settings.issuer_uri || "",
+    idpImageURL: formData.sso_settings.idp_image_url || "",
+    metadata: formData.sso_settings.metadata || "",
+    metadataURL: formData.sso_settings.metadata_url || "",
+    enableSSOIDPLogin: formData.sso_settings.enable_sso_idp_login || false,
+    // SMTP options
+    enableSMTP: formData.smtp_settings.enable_smtp || false,
+    smtpSenderAddress: formData.smtp_settings.sender_address || "",
+    smtpServer: formData.smtp_settings.server || "",
+    smtpPort: formData.smtp_settings.port,
+    smtpEnableSSLTLS: formData.smtp_settings.enable_ssl_tls || false,
+    smtpAuthenticationType: formData.smtp_settings.authentication_type || "",
+    smtpUsername: formData.smtp_settings.user_name || "",
+    smtpPassword: formData.smtp_settings.password || "",
+    smtpAuthenticationMethod:
+      formData.smtp_settings.authentication_method || "",
+    // Global agent options
+    agentOptions: yaml.dump(formData.agent_options) || {},
+    // Host status webhook
+    enableHostStatusWebhook:
+      formData.webhook_settings.host_status_webhook
+        .enable_host_status_webhook || false,
+    hostStatusWebhookDestinationURL:
+      formData.webhook_settings.host_status_webhook.destination_url || "",
+    hostStatusWebhookHostPercentage:
+      formData.webhook_settings.host_status_webhook.host_percentage ||
+      undefined,
+    hostStatusWebhookDaysCount:
+      formData.webhook_settings.host_status_webhook.days_count || undefined,
+    // Usage statistics
+    enableUsageStatistics: formData.server_settings.enable_analytics,
+    // Advanced options
+    domain: formData.smtp_settings.domain || "",
+    verifySSLCerts: formData.smtp_settings.verify_ssl_certs || false,
+    enableStartTLS: formData.smtp_settings.enable_start_tls,
+    enableHostExpiry:
+      formData.host_expiry_settings.host_expiry_enabled || false,
+    hostExpiryWindow: formData.host_expiry_settings.host_expiry_window || 0,
+    disableLiveQuery: formData.server_settings.live_query_disabled || false,
+  });
+
+  const {
+    orgName,
+    orgLogoURL,
+    serverURL,
+    enableSSO,
+    idpName,
+    entityID,
+    issuerURI,
+    idpImageURL,
+    metadata,
+    metadataURL,
+    enableSSOIDPLogin,
+    enableSMTP,
+    smtpSenderAddress,
+    smtpServer,
+    smtpPort,
+    smtpEnableSSLTLS,
+    smtpAuthenticationType,
+    smtpUsername,
+    smtpPassword,
+    smtpAuthenticationMethod,
+    agentOptions,
+    enableHostStatusWebhook,
+    hostStatusWebhookDestinationURL,
+    hostStatusWebhookHostPercentage,
+    hostStatusWebhookDaysCount,
+    enableUsageStatistics,
+    domain,
+    verifySSLCerts,
+    enableStartTLS,
+    enableHostExpiry,
+    hostExpiryWindow,
+    disableLiveQuery,
+  } = iterateFormData;
```

Comment:
Right now these are all destructured to type `any` because of the `useState`. If you provide an interface to `useState` all of these will destructure to the correct types, which should make  typescript a lot more useful in this file.